errors/error.go

```diff
@@ -261,112 +261,58 @@ func (e *Error) Detailed() string {
-	return Is(e, target)
-}
-
-// IsKind tells if err is of kind k.
-// It returns false if err is nil or not an *errors.Error.
-// It also recursively checks if any underlying error is of kind k.
-func IsKind(err error, k Kind) bool {
-	if err == nil {
-		return false
-	}
-	e, ok := err.(*Error)
-	if !ok {
-		return false
-	}
-	if e.Kind != "" && e.Kind == k {
-		return true
-	}
-	return IsKind(e.Err, k)
-}
-
-// Is tells if err (or any of its underlying errors) matches target.
-// It works with any error but when comparing with an errors.Error type it will try to match the following fields:
+// Is tells if err matches the target error.
+// The target error must be of errors.Error type and it will try to match the following fields:
 // - Kind
 // - Description
 // - Stack
 // - FileRange
-// - Err
 // Any fields absent (empty) on the target error are ignored even if they exist on err (partial match).
-// If the target error is no an error.Error it fallbacks to standard errors.Is().
-func Is(err, target error) bool {
-	if (err == nil) != (target == nil) {
-		return false
-	}
-	if target == nil {
-		return err == target
-	}
-	e, ok := err.(*Error)
-	if !ok {
-		return errors.Is(err, target)
-	}
+func (e *Error) Is(target error) bool {
 	t, ok := target.(*Error)
 	if !ok {
-		// target is not *Error so it's already different.
-		// Let's check the underlying error.
-		if e.Err != nil {
-			return Is(e.Err, target)
-		}
 		return false
 	}
 
-	// both are *Error
-
-	if t.Kind != "" && !IsKind(e, t.Kind) {
+	if t.Kind != "" && e.Kind != t.Kind {
 		return false
 	}
 
-	if t.Description != "" && !hasDesc(e, t.Description) {
+	if t.Description != "" && e.Description != t.Description {
 		return false
 	}
 
-	if t.Stack != nil && !hasSameStack(e, t.Stack) {
+	if t.Stack != nil && !equalStack(e.Stack, t.Stack) {
 		return false
 	}
 
 	if !t.FileRange.Empty() && e.FileRange != t.FileRange {
 		return false
 	}
 
-	if t.Err != nil {
-		return Is(e.Err, t.Err)
-	}
-
 	return true
 }
 
-func hasDesc(err error, desc string) bool {
-	e, ok := err.(*Error)
-	if !ok {
-		return false
-	}
+// Unwrap returns the wrapped error, if there is any.
+// Returns nil if there is no wrapped error.
+func (e *Error) Unwrap() error {
+	return e.Err
+}
 
-	if e.Description == desc {
-		return true
-	}
-	if e.Err != nil {
-		return hasDesc(e.Err, desc)
-	}
-	return false
+// IsKind tells if err is of kind k.
+// It is a small wrapper around calling errors.Is(err, errors.Kind(k)).
+func IsKind(err error, k Kind) bool {
+	return Is(err, E(k))
 }
 
-// hasSameStack recursively check if err or any of its underlying errors has the
-// provided stack set.
-func hasSameStack(err error, stack StackMeta) bool {
-	e, ok := err.(*Error)
-	if !ok {
-		return false
-	}
-	if equalStack(e.Stack, stack) {
-		return true
-	}
-	if e.Err != nil {
-		return hasSameStack(e.Err, stack)
-	}
-	return false
+// Is is just an alias to Go stdlib errors.Is
+func Is(err, target error) bool {
```

Comment:
these could be useful to avoid importing two errors package, but not 100% sure they are a good idea. WDYT @i4ki ?

Reply:
yeah It makes sense!